paper-api/src/main/java/org/bukkit/entity/Player.java

```diff
@@ -3892,4 +3893,38 @@ default boolean isChunkSent(@NotNull org.bukkit.Chunk chunk) {
      */
     void sendEntityEffect(org.bukkit.@NotNull EntityEffect effect, @NotNull Entity target);
     // Paper end - entity effect API
+
+    /**
+     * Gives the player the items following full vanilla logic,
+     * making the player drop the items that did not fit into
+     * the inventory. This method will update the count of items
+     * to the amount that did not fit in the inventory.
+     *
+     * @param items the items to give, their count will be mutated
+     * @return spawned item entities
+     */
+    @NotNull List<@NotNull Item> give(@NotNull ItemStack @NotNull... items);
```

Comment:
Just a note, this might wanna be a ItemStack first, ItemStack... other, but we decided against that because itwould make interop with getContents() methods that yield ItemStack[] a lot worse so, may people call this without any items I guess xD